R/utils.R

```diff
@@ -19,60 +23,29 @@
 #' )
 #' assay(se)
 #' assay(relativeAb(se))
+#' 
+#' 
+#' utility function for relativeAb to return counts per million
+#'
+#' @description
+#' This function calculates the relative abundance of each feature in a dataset,
+#' expressed as counts per million (CPM) 
+#' 
+#' @param normalize Logical value indicating whether counts should be normalized
+#' to counts per million (default is TRUE)
+#' 
+#' 
 #'
 #' @export
 relativeAb <- function(se, assay = 1L) {
-  
   assay_data <- assay(se, i = assay)
-  
-  
   csums <- colSums(assay_data)
-  
-  
-  cat("")
-  print(assay_data)
-  
-  
-  res <- assay_data / csums * 1e6
-  
-  
-  cat("")
-  print(res)
-  
-  
+  div <- matrix(rep(csums, each = nrow(assay_data)), ncol = ncol(assay_data))
```

Comment:
Can you explain why you divide by `div` defined on line 43 instead of by `csums` as before? You might be right, I just don't understand the change!

Reply:
I think the matrix div has the same dimension as assay_data